pkg/kv/kvserver/loqrecovery/storage.go

```diff
@@ -0,0 +1,187 @@
+// Copyright 2022 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package loqrecovery
+
+import (
+	"io"
+	"io/fs"
+	"path/filepath"
+	"regexp"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/kv/kvserver/loqrecovery/loqrecoverypb"
+	"github.com/cockroachdb/cockroach/pkg/util/protoutil"
+	"github.com/cockroachdb/errors"
+	"github.com/cockroachdb/pebble/vfs"
+)
+
+const RECOVERY_PLAN_DIR = "recovery-plans"
+
+type PlanStorage struct {
+	path string
+	fs   vfs.FS
+}
+
+type StoredPlan struct {
+	Name         string
+	CreationTime time.Time
+}
+
+func (p StoredPlan) Empty() bool {
+	return len(p.Name) == 0
+}
+
+// NewPlanStore creates a plan store for recovery.
+func NewPlanStore(path string, fs vfs.FS) PlanStorage {
```

Comment:
Why do we pass in `path` here? Isn't it already available via `fs.PathBase`, and shouldn't we use relative paths for everything below it?

Reply:
I don't think that is correct. fs.PathBase() and fs.PathDir() are similar to unix `basename` / `dirname` e.g. they manipulate provided path. And vfs itself is not aware of its directory as defaultFS is an empty struct that just forwards everything to sys calls, while in-memory vfs provides custom functionality. So for in memory case we will have base path as . and for disk case it will point to what --store specifies.

Reply:
I see, my bad, should've looked closer at `vfs.FS`.

Reply:
I looked how we use it and we use empty string as a base path for in-mem which seem to be semantically the same as `.`. I think I'll change that to empty string for consistency with pebble.